pkg/apis/eks.cattle.io/v1/types.go

```diff
@@ -33,63 +33,111 @@ type EKSClusterConfig struct {
 
 // EKSClusterConfigSpec is the spec for a EKSClusterConfig resource
 type EKSClusterConfigSpec struct {
-	AmazonCredentialSecret string            `json:"amazonCredentialSecret"`
-	DisplayName            string            `json:"displayName" norman:"noupdate"`
-	Region                 string            `json:"region" norman:"noupdate"`
-	Imported               bool              `json:"imported" norman:"noupdate"`
-	KubernetesVersion      *string           `json:"kubernetesVersion" norman:"pointer"`
-	Tags                   map[string]string `json:"tags"`
-	SecretsEncryption      *bool             `json:"secretsEncryption" norman:"noupdate"`
-	KmsKey                 *string           `json:"kmsKey" norman:"noupdate,pointer"`
-	PublicAccess           *bool             `json:"publicAccess"`
-	PrivateAccess          *bool             `json:"privateAccess"`
-	EBSCSIDriver           *bool             `json:"ebsCSIDriver"`
-	PublicAccessSources    []string          `json:"publicAccessSources"`
-	LoggingTypes           []string          `json:"loggingTypes"`
-	Subnets                []string          `json:"subnets" norman:"noupdate"`
-	SecurityGroups         []string          `json:"securityGroups" norman:"noupdate"`
-	ServiceRole            *string           `json:"serviceRole" norman:"noupdate,pointer"`
-	NodeGroups             []NodeGroup       `json:"nodeGroups"`
+	// AmazonCredentialSecret is the name of the secret containing the Amazon credentials.
+	AmazonCredentialSecret string `json:"amazonCredentialSecret"`
+	// DisplayName is the name of the cluster to be displayed in the UI.
+	DisplayName string `json:"displayName" norman:"noupdate"`
+	// Region is the AWS region to create the cluster in.
+	Region string `json:"region" norman:"noupdate"`
+	// Imported is true if the cluster was imported.
+	Imported bool `json:"imported" norman:"noupdate"`
+	// KubernetesVersion is the version of Kubernetes to use.
+	KubernetesVersion *string `json:"kubernetesVersion" norman:"pointer"`
```

Comment:
Normally with kubebuilder/controller runtime a pointer indicates that a field is optional. If thats the case for the pointer values here we can `+optional`